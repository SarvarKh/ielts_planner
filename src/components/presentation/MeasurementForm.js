import { CircularProgressbar } from 'react-circular-progressbar';
import React, { useState } from 'react';
import '../../circle-styling.css';

const MeasurementForm = ({exam}) => {
    const [cir_value, setCir_value] = useState(0);

    const setScoreValue = (e) => {
        setCir_value(parseFloat(e.target.value).toFixed(1));
    }

    return (
        <>
            <CircularProgressbar value={cir_value} maxValue={9} text={`${cir_value}`} />
            <div className="tooltip">
                <span className="tooltiptext">Valid only for decimals between 0.0 and 9.0</span>
                <label htmlFor={exam}>Add {exam}</label>
                <input
                    type="number"
                    step=".5"
                    min="0" max="9"
                    name={exam}
                    placeholder="0"
                    required
                    onChange={(e) => setScoreValue(e)}
                />
            </div>
        </>
    )
}

export default MeasurementForm;