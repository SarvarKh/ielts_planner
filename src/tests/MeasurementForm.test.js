import React from 'react';
import renderer from 'react-test-renderer';
import MeasurementForm from '../components/presentation/MeasurementForm';
import { BrowserRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer.create(
  <BrowserRouter>
    <MeasurementForm />
  </BrowserRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
