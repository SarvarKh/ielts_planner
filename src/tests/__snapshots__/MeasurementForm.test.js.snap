// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <svg
    className="CircularProgressbar "
    data-test-id="CircularProgressbar"
    style={Object {}}
    viewBox="0 0 100 100"
  >
    <path
      className="CircularProgressbar-trail"
      d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
      fillOpacity={0}
      strokeWidth={8}
      style={
        Object {
          "strokeDasharray": "289.02652413026095px 289.02652413026095px",
          "strokeDashoffset": "0px",
        }
      }
    />
    <path
      className="CircularProgressbar-path"
      d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
      fillOpacity={0}
      strokeWidth={8}
      style={
        Object {
          "strokeDasharray": "289.02652413026095px 289.02652413026095px",
          "strokeDashoffset": "289.02652413026095px",
        }
      }
    />
    <text
      className="CircularProgressbar-text"
      style={Object {}}
      x={50}
      y={50}
    >
      0
    </text>
  </svg>
  <div
    className="tooltip"
  >
    <span
      className="tooltiptext"
    >
      Valid only for decimals between 0.0 and 9.0
    </span>
    <label />
    <input
      max="9"
      min="0"
      onChange={[Function]}
      placeholder="Enter score"
      required={true}
      step=".5"
      type="number"
    />
  </div>
</div>
`;
